solidity/contracts/igps/GasOverheadIgp.sol

```diff
@@ -0,0 +1,179 @@
+
+    /// @notice Destination domain => overhead gas amount on that domain.
+    mapping(uint32 => uint256) public destinationGasOverhead;
+
+    // ============ Upgrade Gap ============
+
+    // gap for upgrade safety
+    uint256[48] private __GAP;
+
+    // ============ Events ============
+
+    /**
+     * @notice Emitted when the innerIgp is set.
+     * @param innerIgp The new innerIgp.
+     */
+    event InnerIgpSet(address innerIgp);
+
+    /**
+     * @notice Emitted when an entry in the destinationGasOverhead mapping is set.
+     * @param domain The destination domain.
+     * @param gasOverhead The gas overhead amount on that domain.
+     */
+    event DestinationGasOverheadSet(uint32 indexed domain, uint256 gasOverhead);
+
+    // ============ Constructor ============
+
+    constructor(address _innerIgp) {
+        initialize(_innerIgp); // allows contract to be used without proxying
+    }
+
+    // ============ Initializers ============
+
+    /**
+     * @notice Initializes the contract.
+     * @param _innerIgp The innerIgp.
+     */
+    function initialize(address _innerIgp) public initializer {
+        __Ownable_init();
+
+        _setInnerIgp(_innerIgp);
+    }
+
+    // ============ External Functions ============
+
+    /**
+     * @notice Adds the stored destinationGasOverhead to the _gasAmount and forwards the
+     * call to the innerIgp's `payForGas` function.
+     * @param _messageId The ID of the message to pay for.
+     * @param _destinationDomain The domain of the message's destination chain.
+     * @param _gasAmount The amount of destination gas to pay for. This should not
+     * consider any gas that is accounted for in the stored destinationGasOverhead.
+     * @param _refundAddress The address to refund any overpayment to.
+     */
+    function payForGas(
+        bytes32 _messageId,
+        uint32 _destinationDomain,
+        uint256 _gasAmount,
+        address _refundAddress
+    ) external payable {
+        innerIgp.payForGas{value: msg.value}(
+            _messageId,
+            _destinationDomain,
+            destinationGasAmount(_destinationDomain, _gasAmount),
+            _refundAddress
+        );
+    }
+
+    /**
+     * @notice Sets destination gas overheads for multiple domains.
+     * @dev Only callable by the owner.
+     * @dev _domains.length and _gasOverheads.length must be the same.
+     * @param _domains The destination domains to set gas overheads for.
+     * @param _gasOverheads The gas overheads for each corresponding domain in _domains.
+     */
+    function setDestinationGasOverheads(
+        uint32[] calldata _domains,
+        uint256[] calldata _gasOverheads
+    ) external onlyOwner {
+        require(
+            _domains.length == _gasOverheads.length,
+            "Domain and gas overhead length mismatch"
+        );
+
+        for (uint256 i; i < _domains.length; i++) {
+            _setDestinationGasOverhead(_domains[i], _gasOverheads[i]);
+        }
```

Comment:
```suggestion
    ```suggestion
    struct DomainConfig {
        uint32 domain;
        uint256 gasOverhead;
    }
    
    function setDestinationGasOverheads(
        DomainConfig[] configs
    ) external onlyOwner {
        for (uint256 i; i < _domains.length; i++) {
            _setDestinationGasOverhead(configs[i]);
        }
```